render/wlr_renderer.c

```diff
@@ -251,6 +255,7 @@ bool wlr_renderer_init_wl_display(struct wlr_renderer *r,
 }
 
 struct wlr_renderer *wlr_renderer_autocreate_with_drm_fd(int drm_fd) {
+#if WLR_HAS_GLES2_RENDERER
```

Comment:
Maybe we should import `IS_ENABLED` from the linux kernel.

Reply:
Not a fan of magic kernel macros. In particular, `IS_ENABLED` seems to be using a lot of black magic.

Reply:
I found the explanation in [Zephyr](https://github.com/zephyrproject-rtos/zephyr/blob/89a90963343437378a462aa64431b2283b7c397a/include/sys/util_internal.h#L18) good. But it totally looks like black magic. This would also require changes to the conditional include of the header. The advantage is that all code is compiled by the compiler and bit rot of conditional code paths can be avoided.